### ISSUE TITLE
Panic in Skaffold 1.39.0 when rebuilding with new tag

### ISSUE THREAD

**Issue by armsnyder** [GoogleContainerTools/skaffold#7587]
### Regression Note

I'm able to reproduce this panic in 1.39.0, but not in 1.38.0, to confirm it is a regression.

### Expected behavior

I expect to be able to run `skaffold build` twice, with different tag configuration each time, in order to push multiple tags for the same image.

### Actual behavior

Skaffold panics while attempting to re-tag.

```
➜  CONTAINER_SEM_VER=9999.9.10 skaffold build
Generating tags...
 - redacted.dkr.ecr.us-east-1.amazonaws.com/redacted/backend -> dkr.ecr.us-east-1.amazonaws.com/backend:9999.9.10
Checking cache...
 - redacted.dkr.ecr.us-east-1.amazonaws.com/backend: Found. Tagging
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.needsRemoteTagging.Tag({{0xc00105dbc0, 0x40}, {0xc00096c300, 0x54}, {0xc00104da90, 0x47}}, {0x1, 0x1}, 0xc000a44580, {0xc000dcdf20})
        /tmpfs/src/github/skaffold/pkg/skaffold/build/cache/details.go:94 +0x257
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.(*cache).Build(0xc000a44580, {0x6965a40, 0xc001089230}, {0x690e580, 0xc000a484b0}, 0xc001089290, {0xc0011133a0, 0x2, 0x8}, {0xc000dcdf20}, ...)
        /tmpfs/src/github/skaffold/pkg/skaffold/build/cache/retrieve.go:85 +0xb58
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*Builder).Build(0xc0001dcf00, {0x6965998, 0xc000a791c0}, {0x690e580, 0xc0003aa3c0}, {0xc0011133a0, 0x2, 0x2})
        /tmpfs/src/github/skaffold/pkg/skaffold/runner/build.go:105 +0x445
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doBuild.func1({0x69c24a0, 0xc0001dcf00}, {0xc00044d860, 0xd, 0x63fa446})
        /tmpfs/src/github/skaffold/cmd/skaffold/app/cmd/build.go:72 +0x104
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.withRunner({0x6965998, 0xc000a791c0}, {0x690e580, 0xc0003aa3c0}, 0xc0007fbae0)
        /tmpfs/src/github/skaffold/cmd/skaffold/app/cmd/runner.go:55 +0xb4
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doBuild({0x6965998, 0xc000a791c0}, {0x690e580, 0xc0003aa3c0})
        /tmpfs/src/github/skaffold/cmd/skaffold/app/cmd/build.go:71 +0xa9
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.(*builder).NoArgs.func1(0x45b7a00, {0xc000b27b80, 0xc000b27b58, 0x0})
        /tmpfs/src/github/skaffold/cmd/skaffold/app/cmd/commands.go:138 +0x66
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.applyPostRunHooks.func1(0xc0004e4b00, {0x7f727a0, 0x0, 0x0})
        /tmpfs/src/github/skaffold/cmd/skaffold/app/cmd/commands.go:185 +0x2f
github.com/spf13/cobra.(*Command).execute(0xc0004e4b00, {0x7f727a0, 0x0, 0x0})
        /Users/kbuilder/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc0007c8000)
        /Users/kbuilder/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).ExecuteContextC(...)
        /Users/kbuilder/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:911
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run({0x690d600, 0xc00018e008}, {0x690d600, 0xc00018e010})
        /tmpfs/src/github/skaffold/cmd/skaffold/app/skaffold.go:50 +0x2ad
main.main()
        /tmpfs/src/github/skaffold/cmd/skaffold/skaffold.go:49 +0x1e5
```

### Information

- Skaffold version: `v1.39.0`
- Operating system: `MacOS 11.6.7 (amd64)`
- Installed via: `Homebrew`
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta19
kind: Config
build:
  local:
    useBuildkit: true
  artifacts:
    - image: redacted.dkr.ecr.us-east-1.amazonaws.com/backend
  build:
      tagPolicy:
        envTemplate:
          template: "{{.CONTAINER_SEM_VER}}"
```

### Steps to reproduce the behavior

1. `CONTAINER_SEM_VER=1.1.1 skaffold build` (succeeds)
2. `CONTAINER_SEM_VER=2.2.2 skaffold build` (panics)

### Notable debug log output

I found this section of the debug log output to be notable because the platforms detected is `""`, which would cause the index out of range panic when the platforms slice is indexed.

```
DEBU[0001] setting Docker user agent to skaffold-v1.39.0  subtask=-1 task=DevLoop
DEBU[0001] CLI platforms provided: ""                    subtask=-1 task=DevLoop
DEBU[0001] platforms detected from active kubernetes cluster nodes: ""  subtask=-1 task=DevLoop
DEBU[0001] platforms selected for artifact "redacted.dkr.ecr.us-east-1.amazonaws.com/backend": ""  subtask=-1 task=DevLoop
DEBU[0001] Using builder: local                          subtask=-1 task=DevLoop
DEBU[0001] push value not present in NewBuilder, defaulting to true because cluster.PushImages is true  subtask=-1 task=DevLoop
INFO[0001] build concurrency first set to 1 parsed from *local.Builder[0]  subtask=-1 task=DevLoop
INFO[0001] final build concurrency value is 1            subtask=-1 task=DevLoop
Generating tags...
```

**Comment by yckbilly1929** [comment]
same for 1.39.1
waiting for https://github.com/GoogleContainerTools/skaffold/pull/7592 to be merged 👍

**Comment by gsquared94** [comment]
there seems to be some issue with skaffold installed using `brew`. Can you try downloading the binary directly using:
```
curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/v1.39.1/skaffold-darwin-arm64 && chmod +x skaffold && sudo mv skaffold /usr/local/bin
```

---

I can repro the issue on `v1.39.0`:
```
curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/v1.39.0/skaffold-darwin-arm64 && chmod +x skaffold && sudo mv skaffold /usr/local/bin
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 71.7M  100 71.7M    0     0  7150k      0  0:00:10  0:00:10 --:--:-- 9591k
❯ skaffold build --tag=foo1 --default-repo=gcr.io/k8s-skaffold
Generating tags...
 - skaffold-example -> gcr.io/k8s-skaffold/skaffold-example:foo1
Checking cache...
 - skaffold-example: Found. Tagging
panic: runtime error: index out of range [0] with length 0
...
```
Then I tried `v1.39.1` installed with `brew`:
```
❯ sudo rm $(which skaffold)
❯ brew install skaffold
==> Downloading https://ghcr.io/v2/homebrew/core/skaffold/manifests/1.39.1
...
❯ skaffold version
v1.39.1
❯ skaffold build --tag=foo1 --default-repo=gcr.io/k8s-skaffold
Generating tags...
 - skaffold-example -> gcr.io/k8s-skaffold/skaffold-example:foo1
Checking cache...
 - skaffold-example: Found. Tagging
panic: runtime error: index out of range [0] with length 0
...
```
Then I tried installing the `v1.39.1` binary directly:
```
❯ brew remove skaffold
Uninstalling /opt/homebrew/Cellar/skaffold/1.39.1... (8 files, 80.7MB)
❯ curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/v1.39.1/skaffold-darwin-arm64 && chmod +x skaffold && sudo mv skaffold /usr/local/bin
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 71.7M  100 71.7M    0     0  29.4M      0  0:00:02  0:00:02 --:--:-- 29.4M
❯ skaffold build --tag=foo1 --default-repo=gcr.io/k8s-skaffold
Generating tags...
 - skaffold-example -> gcr.io/k8s-skaffold/skaffold-example:foo1
Checking cache...
 - skaffold-example: Found. Tagging

❯ skaffold build --tag=foo2 --default-repo=gcr.io/k8s-skaffold
Generating tags...
 - skaffold-example -> gcr.io/k8s-skaffold/skaffold-example:foo2
Checking cache...
 - skaffold-example: Found. Tagging
```

**Comment by yxfeng-tw** [comment]
try to remove cache resolved the issue

`rm -rf ~/.skaffold/cache`

**Comment by gsquared94** [comment]
this has been fixed in the [v1.39.2 release](https://github.com/GoogleContainerTools/skaffold/releases/tag/v1.39.2)
